spec/support/acceptance/shared_contexts.rb

```diff
@@ -26,3 +26,19 @@
     IncidentHelpers.delete_incidents(query)
   end
 end
+
+RSpec.shared_context 'corrective change' do |noop = false|
+  before(:each) do
+    # For a change to be considered corrective, the resource needs to be managed and in
+    # the correct state at least one time, and then a drift from that state corrected.
+    # This setup manages a file and then drifts its state. The noop added to the
+    # resource will still mark the report as noop_pending.
+    set_sitepp_content(to_manifest(declare('file', '/tmp/test', 'content' => 'hello')))
```

Comment:
Nit: `apply_manifest`'ing this part would be faster than a full-blown server run.